ui/server/src/main/scala/pl/touk/nussknacker/ui/db/DatabaseInitializer.scala

```diff
@@ -24,35 +20,29 @@ object DatabaseServer {
   case class Config(dbFilePath:String, dbName:String, user:String, password:String, port:Option[Int], enabled: Option[Boolean])
 }
 
-class DatabaseInitializer(dbConfig: DbConfig) {
-  def initDatabase(): Unit = {
-    migrateIfNeeded(dbConfig)
-  }
-
-  private def migrateIfNeeded(dbConfig: DbConfig) = {
-    val flyway = Flyway.configure()
-      .dataSource(new DatabaseDataSource(dbConfig.db))
-      .baselineOnMigrate(true)
+object DatabaseInitializer {
 
-    val flywayWithDbProfile = dbConfig.driver match {
-      case HsqldbProfile => flyway.locations("db/migration/hsql", "db/migration/common")
-      case PostgresProfile => flyway.locations("db/migration/postgres", "db/migration/common")
-    }
-    flywayWithDbProfile.load().migrate()
+  def initDatabase(path: String, config: Config): Unit = {
+    import net.ceedubs.ficus.Ficus._
+    val configDb = config.getConfig(path)
+    DatabaseInitializer.initDatabase(
+      configDb.as[String]("url"),
+      configDb.as[String]("user"),
+      configDb.as[String]("password"),
+      configDb.getAs[String]("schema"))
   }
-}
-
-class DatabaseDataSource(db: JdbcBackend.Database) extends DataSource {
-  private val conn = db.createSession().conn
 
-  override def getConnection: Connection = conn
-  override def getConnection(username: String, password: String): Connection = conn
-  override def unwrap[T](iface: Class[T]): T = conn.unwrap(iface)
-  override def isWrapperFor(iface: Class[_]): Boolean = conn.isWrapperFor(iface)
-
-  override def setLogWriter(out: PrintWriter): Unit = ???
-  override def getLoginTimeout: Int = ???
-  override def setLoginTimeout(seconds: Int): Unit = ???
-  override def getParentLogger: Logger = ???
-  override def getLogWriter: PrintWriter = ???
+  def initDatabase(url: String, user: String, password: String, schema: Option[String] = None): Unit =
+    Flyway
+      .configure()
+      .dataSource(url, user, password)
+      .schemas(schema.toArray: _*)
+      .baselineOnMigrate(true)
+      .locations(
```

Comment:
I think we should raise some exception when url doesn't match to these two patterns.

Reply:
I haven't found any generic purpose exception class throughout the source code that would pretty printed to the user in case of wrong configuration. I few places an IllegalArgumentException is thrown. I doubt it is a proper exception to be communicated to the user. Should I use it?

Reply:
IllegalArgumentException seems to be good choice ;)

Reply:
Done.